src/views/MovieListPage.tsx

```diff
@@ -0,0 +1,158 @@
+import { AnimatePresence } from 'framer-motion';
+
+import { useEffect, useState } from 'react';
+
+import SearchForm from '@components/SearchForm';
+import GenreSelect from '@components/GenreSelect';
+import MovieTile from '@components/MovieTile';
+import MovieDetails from '@components/MovieDetails';
+import SortControl from '@components/SortControl';
+import { DeleteModal, MovieModal } from '@components/modals';
+
+import { Genres, IMovieDetails, IOption } from '../types';
+import { MOVIE_MODAL, DELETE_MODAL } from '../constants';
+import styles from './styles.module.scss';
+
+interface IMovieListPage {
+  movieDetailsHandler: () => void;
+}
+
+const MovieListPage = ({ movieDetailsHandler }: IMovieListPage) => {
+  const [moviesList, setMoviesList] = useState<IMovieDetails[]>([]);
+  const [query, setQuery] = useState({
+    search: '',
+    filter: '',
+    sortBy: 'release_date',
+  });
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const searchQuery = query.search ? '&search=' + query.search : '';
+      const filterQuery = query.filter ? '&filter=' + query.filter : '';
+      const queryUrl = `?limit=24&searchBy=title&sortOrder=asc&sortBy=${query.sortBy}${searchQuery}${filterQuery}`;
+      try {
+        const result = await fetch(`http://localhost:4000/movies${queryUrl}`);
+        const { data } = await result.json();
+        setMoviesList(data);
+      } catch (err) {
+        setMoviesList([]);
+      }
+    };
+    setIsLoading(true);
+    fetchData();
+    setTimeout(() => setIsLoading(false), 200);
```

Comment:
`setIsLoading` handler should be placed after `setMoviesList`
Another popular solution - custom hook useStateCallback 
[link](https://stackoverflow.com/a/61842546)